specification/frontdoor/resource-manager/Microsoft.Network/preview/2018-08-01-preview/frontdoor.json

```diff
@@ -1808,6 +1826,67 @@
             "name": "FrontDoorEnabledState",
             "modelAsString": true
           }
+        },
+        "redirectConfiguration": {
+          "description": "A reference to the redirect routing configuration. ",
+          "$ref": "./network.json#/definitions/RedirectConfiguration"
+        }
+      }
+    },
+    "RedirectConfiguration": {
+      "description": "The configuration for a redirect routing rule. This object is needed only if the type property of RoutingRule is set to Redirect. ",
+      "type": "object",
+      "properties": {
+        "redirectType": {
+          "description": "The redirect type the rule will use when redirecting traffic.",
+          "type": "string",
+          "enum": [
+            "Moved(301)",
+            "Found(302)",
+            "TemporaryRedirect(307)",
+            "PermanentRedirect(308)"
+          ],
+          "x-ms-enum": {
+            "name": "FrontDoorRedirectProtocol",
+            "modelAsString": true
+          }
+        },
+        "destinationProtocol": {
+          "description": "The protocol of the destination where the traffic is forwarded to",
+          "type": "string",
+          "enum": [
+            "MatchRequest",
+            "Http",
+            "Https"
+          ],
+          "x-ms-enum": {
+            "name": "FrontDoorDestinationProtocol",
+            "modelAsString": true
+          }
+        },
+        "destinationHost": {
+          "type": "string",
+          "description": "If left blank, then we will use the incoming host as the destination host."
+        },
+        "destinationPath": {
```

Comment:
nit: you can use pattern specifiers to enforce this. Not blocking on this.
An example [here](https://github.com/Azure/azure-rest-api-specs/blob/32b0d873aa851d456dfde7d6ba1d89ff33f897f0/specification/batch/resource-manager/Microsoft.Batch/stable/2015-12-01/BatchManagement.json#L52)